src/components/NewCommentForm.tsx

```diff
@@ -1,103 +1,236 @@
+              id="comment-author-name"
+              placeholder="Name Surname"
+              className={cn('input',
+                {
+                  'is-danger': !nameTrim && checkFieldName,
+                })}
+              value={name}
+              onChange={(event) => {
+                setCheckFieldName(false);
+                setName(event.target.value);
+              }}
+            />
 
-      <div className="field" data-cy="EmailField">
-        <label className="label" htmlFor="comment-author-email">
-          Author Email
-        </label>
-
-        <div className="control has-icons-left has-icons-right">
-          <input
-            type="text"
-            name="email"
-            id="comment-author-email"
-            placeholder="email@test.com"
-            className="input is-danger"
-          />
-
-          <span className="icon is-small is-left">
-            <i className="fas fa-envelope" />
-          </span>
-
-          <span
-            className="icon is-small is-right has-text-danger"
-            data-cy="ErrorIcon"
-          >
-            <i className="fas fa-exclamation-triangle" />
-          </span>
+            <span className="icon is-small is-left">
+              <i className="fas fa-user" />
+            </span>
+            {!nameTrim && checkFieldName && (
+              <span
+                className="icon is-small is-right has-text-danger"
+                data-cy="ErrorIcon"
+              >
+                <i className="fas fa-exclamation-triangle" />
+              </span>
+            )}
+          </div>
+          {!nameTrim && checkFieldName && (
+            <p className="help is-danger" data-cy="ErrorMessage">
+              Name is required
+            </p>
+          )}
         </div>
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Email is required
-        </p>
-      </div>
+        <div className="field" data-cy="EmailField">
+          <label className="label" htmlFor="comment-author-email">
+            Author Email
+          </label>
 
-      <div className="field" data-cy="BodyField">
-        <label className="label" htmlFor="comment-body">
-          Comment Text
-        </label>
-
-        <div className="control">
-          <textarea
-            id="comment-body"
-            name="body"
-            placeholder="Type comment here"
-            className="textarea is-danger"
-          />
+          <div className="control has-icons-left has-icons-right">
+            <input
+              type="text"
+              name="email"
+              id="comment-author-email"
+              placeholder="email@test.com"
+              className={cn('input',
+                {
+                  'is-danger': !emailTrim && checkFieldEmail,
+                })}
+              value={email}
+              onChange={(event) => {
+                setCheckFieldEmail(false);
+                setEmail(event.target.value);
+              }}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**